Doc/faq/design.rst

```diff
@@ -444,12 +444,12 @@ far) under most circumstances, and the implementation is simpler.
 
 Dictionaries work by computing a hash code for each key stored in the dictionary
 using the :func:`hash` built-in function.  The hash code varies widely depending
-on the key and a per-process seed; for example, "Python" could hash to
--539294296 while "python", a string that differs by a single bit, could hash
-to 1142331976.  The hash code is then used to calculate a location in an
+on the key and a per-process seed; for example, ``'Python'`` could hash to
+``-539294296`` while ``'python'``, a string that differs by a single bit, could hash
+to ``1142331976``.  The hash code is then used to calculate a location in an
 internal array where the value will be stored.  Assuming that you're storing
 keys that all have different hash values, this means that dictionaries take
-constant time -- O(1), in Big-O notation -- to retrieve a key.
+constant time -- ``O(1)``, in Big-O notation -- to retrieve a key.
```

Comment:
```suggestion
constant time -- O(1), in Big-O notation -- to retrieve a key.
```

Is big-O notation really written as code literals, since it isn't code?

Reply:
That's true, but I would still expect there to be some sort of formatting. Taking for instance `O(log n)` and `O(1)`, I think that since it makes sense for the first to be formatted, the latter should be too, though I'm not very particular on this.

Reply:
If it does need formatting, I'd expect it to be italics, as is general textbook convention (IIRC) for inline mathematical symbology.

Of the 24 instances I found in the CPython docs that matched`\bO\(`, all of which appeared to be valid big-O notation, 3 (in one file) used a code literal, 4 (in another file) used italics, and the other 17 (in 12 files) all used prose. In fact, non-formatted is _required_ for any big-O notation that has an exponent other than 1, a log base other than the default, or any other superscripts or subscripts, as several of them (in two different files) do (e.g. using e.g. `` O(n\ :sup:`2`) ``, to make O(n<sup>2</sup>).

Reply:
Hmm that's a good point. If there were some super/subscripts needed, as well as for instance `O(log n)`, then for consistency's sake it seems like keeping them both non-literal and non-italisized would be best.